pkg/webhooks/kafkatopic_validator_test.go

```diff
@@ -193,6 +193,7 @@ func TestCheckKafkaTopicExist(t *testing.T) {
 	}
 
 	for _, testCase := range testCases {
+		testCase := testCase
```

Comment:
Why is this necessary?

Reply:
Because of how the iteration pointer works in Go together with the capture variables for the lambda function defined on the next line where the iteration pointer is a lambda capture variable and running the tests in a parallel fashion.

With `t.Parallel()` the iteration just fires the subtests but won't be blocked until it finishes, so multiple `t.Run()`s are going to be executed at the same time.
All of those are going to refer to `&testCase.KafkaTopic` for example and when the iteration moves ahead the `testCase` pointer will be updated in place to the new value.
If the iteration is at test case X and executes `t.Run()` then detaches the thread with `t.Parallel()` and the iteration moves ahead to test case Y updating the `testCase` iteration pointer then X's thread executes the the `fieldErrorList, err := kafkaTopicValidator.checkKafka(context.Background(), &testCase.kafkaTopic, cluster)` statement, it will evaluate `&testCase.kafkaTopic` to Y's `testCase.kafkaTopic` value instead of X's.
By declaring another variable inside the iteration to hold the iterator's value, we copy the original content, ensuring the lambda function always calls the iteration-local copy which means using X's values in X's lambda function.
The convention of naming the iteration local variable the same way as the iterator is to ease the developer's work because semantically the iterator changing and the lambda function using the new value of it is merely a Go optimization internal detail the developer doesn't care much about.

Reply:
I had no idea, thanks a lot for explaining!